LabGui.py

```diff
@@ -1292,12 +1316,38 @@ def option_change_debug_state(self):
 
             self.emit(SIGNAL("DEBUG_mode_changed(bool)"), self.DEBUG)
 
+
+    def option_command_window_state(self):
+        """launches or focuses commandline """
+        if self.cmddock:
+            self.cmddock.setFloating(True)
+            self.cmddock.toggleViewAction()
+            self.cmddock.show()
+        else:
+            self.cmdline.show()
+            self.cmdline.update_devices()
+            self.cmdline.raise_()
+
+
     def option_change_interface(self):
         """changes GPIB interface"""
         intf = str(self.sender().text())
         # changes the GPIB interface in the instrument hub so the next time
         # one connects instrument it will be through this interface
+        try:
+            if not isinstance(self.output_file, str): # else, output_file.close() will be run
+                self.stop_DTT() # must stop and DC everything first
```

Comment:
what does DC everything means?

Reply:
typo :P I believe I meant to write "must stop DTT first"

Perhaps there is a better solution to this, such as (also) sending a PyQt signal to stop any running scripts, especially to prevent crashing with any future user widgets that operate independently from DTT